### ISSUE TITLE
Please add Three Rivers District Council

### ISSUE THREAD

**Issue by sbrewster2** [robbrad/UKBinCollectionData#490]
### Name of Council

Three Rivers District Council

### Example Address/Postcode

WD3 7AZ

### Additional Information

From this link https://my.threerivers.gov.uk/en, click Bin Collection Day. Requires a postcode and gives you the next 2 weeks collections. 

### Verification

- [X] I searched for similar issues at https://github.com/robbrad/UKBinCollectionData/issues?q=is:issue and found no duplicates
- [X] I have provided a tested working address/postcode/UPRN with bin collections available
- [X] I understand that this project is run by volunteer contributors and completion depends on numerous factors - even with a request, we cannot guarantee if/when your council will get a script

**Comment by skelt0** [comment]
Data does come back as JSON but a load of session junk to get there. Selinium candidate for sure

<img width="540" alt="image" src="https://github.com/robbrad/UKBinCollectionData/assets/16990151/dea293f7-b2ae-40a3-aee0-33e4196d165c">

**Comment by dp247** [comment]
> Data does come back as JSON but a load of session junk to get there. Selinium candidate for sure
> 
> <img alt="image" width="540" src="https://private-user-images.githubusercontent.com/16990151/291018617-dea293f7-b2ae-40a3-aee0-33e4196d165c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI3Njc1ODksIm5iZiI6MTcwMjc2NzI4OSwicGF0aCI6Ii8xNjk5MDE1MS8yOTEwMTg2MTctZGVhMjkzZjctYjJhZS00MGEzLWFlZTAtMzNlNDE5NmQxNjVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE2VDIyNTQ0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjODU1NmQyNTA3YWJlZGU2ZGU2MTg3NjVjNDQyZjlmNDAyMGJlNTIzYTRmYjY1NTlhMTMxN2ZhOWY0OThmOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-JKlqNQIj0120ygudFgSQ-peYnrCBstlCgjbEYfA1OU">

Could be adapted from all the ones @robbrad just pushed too, its an AchieveForms one again 😄

**Comment by rajp1503** [comment]
Just wondering, what does the AchieveForms mean?? I am a noob and have no clue on this.

**Comment by dp247** [comment]
> Just wondering, what does the AchieveForms mean?? I am a noob and have no clue on this.

It's a kind of forms used a lot in local/central government. We see a lot of them there, they can be kind of hard to parse sometimes.

**Comment by rajp1503** [comment]
> > Just wondering, what does the AchieveForms mean?? I am a noob and have no clue on this.
> 
> It's a kind of forms used a lot in local/central government. We see a lot of them there, they can be kind of hard to parse sometimes.

Ahh typical of local govt.. make it more complicated then it needs to be.. is there no hope of disecting this info? I can get the UPRN number for my property.

**Comment by dp247** [comment]
I'll probably do it some other way sometime

**Comment by robbrad** [comment]
Selenium required:

https://my.threerivers.gov.uk/en/AchieveForms/?mode=fill&consentMessage=yes&form_uri=sandbox-publish://AF-Process-52df96e3-992a-4b39-bba3-06cfaabcb42b/AF-Stage-01ee28aa-1584-442c-8d1f-119b6e27114a/definition.json&process=1&process_uri=sandbox-processes://AF-Process-52df96e3-992a-4b39-bba3-06cfaabcb42b&process_id=AF-Process-52df96e3-992a-4b39-bba3-06cfaabcb42b&noLoginPrompt=1

Same system as CheshireWestAndChesterCouncil and ReigateAndBansteadBoroughCouncil

iframes FTW !

**Comment by rajp1503** [comment]
> Selenium required:
> 
> https://my.threerivers.gov.uk/en/AchieveForms/?mode=fill&consentMessage=yes&form_uri=sandbox-publish://AF-Process-52df96e3-992a-4b39-bba3-06cfaabcb42b/AF-Stage-01ee28aa-1584-442c-8d1f-119b6e27114a/definition.json&process=1&process_uri=sandbox-processes://AF-Process-52df96e3-992a-4b39-bba3-06cfaabcb42b&process_id=AF-Process-52df96e3-992a-4b39-bba3-06cfaabcb42b&noLoginPrompt=1
> 
> Same system as CheshireWestAndChesterCouncil and ReigateAndBansteadBoroughCouncil
> 
> iframes FTW !

Thanks Rob.. does this mean it is an easy enhancement?? Any ETA when this can be implemented??

**Comment by robbrad** [comment]
Hopeful it will be in the next release

**Comment by robbrad** [comment]
@rajp1503 - got to add a few more councils here and this will be released soon
